packages/expo-cli/src/commands/utils/CreateApp.ts

```diff
@@ -0,0 +1,185 @@
+import * as PackageManager from '@expo/package-manager';
+import program from 'commander';
+import fs from 'fs-extra';
+import getenv from 'getenv';
+import yaml from 'js-yaml';
+import ora from 'ora';
+import * as path from 'path';
+import semver from 'semver';
+
+import log from '../../log';
+
+export function validateName(name?: string): string | true {
+  if (typeof name !== 'string' || name === '') {
+    return 'The project name can not be empty.';
+  }
+  if (!/^[a-z0-9@.\-_]+$/i.test(name)) {
+    return 'The project name can only contain URL-friendly characters.';
```

Comment:
```suggestion
    return 'The project name can only contain URL-friendly characters (alphanumeric and @ . -  _)';
```

maybe worth adding some additional info on what characters these are?